### PR TITLE
#872 Make metric test helpers public

### PR DESCRIPTION
**Pull request by AndrewGrachov**
The main purpose of this PR is to provide unified solution for all metric testing, using mock meter. (Both instrumentation and internal). If the concept is ok, It could be extended by such methods as 

CheckSingleRecord

CheckSingleObserver

etc.

and all aggregation test helpers

### CODE REVIEW

**Comment by AndrewGrachov** [comment]
My guess we should move that out of internal package and create something like testHelpers internal package which could be used in plugin instrumentations

File: internal/metric/test_helpers.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/metric"
+)
+
+func ResolveNumberByKind(t *testing.T, kind metric.NumberKind, value float64) metric.Number {
```
Comment (MrAlias) [inline internal/metric/test_helpers.go:14]: Need documentation.

File: internal/metric/test_helpers.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	case metric.Int64NumberKind:
+		return metric.NewInt64Number(int64(value))
+	case metric.Float64NumberKind:
+		return metric.NewFloat64Number(value)
+	}
+	panic("invalid number kind")
+}
+
+// TODO: ADD doc
+func CheckSyncBatches(t *testing.T, ctx context.Context, labels []kv.KeyValue, mock *MeterImpl, nkind metric.NumberKind, mkind metric.Kind, instrument metric.InstrumentImpl, expected ...float64) {
```
Comment (MrAlias) [inline internal/metric/test_helpers.go:26]: Need documentation.

File: internal/metric/test_helpers.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
+package metric
```
Comment (MrAlias) [inline internal/metric/test_helpers.go:1]: Is the end goal to have this eventually live in something like a `go.opentelemetry.io/otel/api/metric/metrictest` package?

**Pull request merged**